Bugzilla/Auth/Login/APIKey.pm

```diff
@@ -39,36 +38,57 @@ sub app_id {
 # be used to authenticate a Web request.
 sub get_login_info {
   my ($self) = @_;
+  my $cgi    = Bugzilla->cgi;
   my $params = Bugzilla->input_params;
   my ($user_id, $login_cookie);
 
-  my $api_key_text = trim(delete $params->{'Bugzilla_api_key'});
-  if (!i_am_webservice() || !$api_key_text) {
-    return {failure => AUTH_NODATA};
-  }
-
-  my $api_key   = Bugzilla::User::APIKey->new({name => $api_key_text});
-  my $remote_ip = remote_ip();
+  # First check for an API key in the header or passed as query params
+  my $api_key_text = trim($cgi->http('X_BUGZILLA_API_KEY'))
+    || trim($params->{'Bugzilla_api_key'});
 
-  if (!$api_key or $api_key->api_key ne $api_key_text) {
+  # Also allow use of OAuth2 bearer tokens to access the API
+  # substr() removes the "Bearer " portion of the header
+  my $oauth_token = trim($cgi->http('Authorization')) || '';
+  $oauth_token = substr($oauth_token, 7) if $oauth_token;
```

Comment:
I'm not sure why this is happening as all the code does is check if this header was present; the contents of `$oauth_token` is ignored here.

If you want to retain the substringing, the following would be clearer and remove the need to document what the substr call is achieving:

`$oauth_token =~ s/^Bearer //i if $oauth_token;`